### DISCUSSION TITLE
Building Go v1 apps with dependences

### DISCUSSION THREAD

**Discussion by suntong** [goreleaser/goreleaser#2265] [Q&A] [answered]
Following up on https://github.com/goreleaser/goreleaser/issues/2149,

For projects without go mod, goreleaser is _working_.

However, when I try on projects without go mod but with dependences, it is failing --
https://github.com/suntong/shuttlebot/runs/2701444157?check_suite_focus=true

```
   • building binaries
      • building                  binary=/home/runner/work/shuttlebot/shuttlebot/dist/shuttlebot_windows_amd64/shuttlebot.exe
      • building                  binary=/home/runner/work/shuttlebot/shuttlebot/dist/shuttlebot_linux_amd64/shuttlebot
      • building                  binary=/home/runner/work/shuttlebot/shuttlebot/dist/shuttlebot_darwin_amd64/shuttlebot
   ⨯ release failed after 0.20s error=failed to build for linux_amd64: main.go:16:2: cannot find package "github.com/caarlos0/env" in any of:
	/opt/hostedtoolcache/go/1.15.12/x64/src/github.com/caarlos0/env (from $GOROOT)
	/home/runner/go/src/github.com/caarlos0/env (from $GOPATH)
logging.go:8:2: cannot find package "github.com/go-kit/kit/log" in any of:
	/opt/hostedtoolcache/go/1.15.12/x64/src/github.com/go-kit/kit/log (from $GOROOT)
	/home/runner/go/src/github.com/go-kit/kit/log (from $GOPATH)
. . .
```

Is it true that dependences is the problem, or something else (e.g., problem only relates to `_darwin_amd64`, but not others like `_linux_amd64` etc)?

Would it be possible that I set my own `GOPATH` and do `go get` myself?

Thx

**Comment by caarlos0** [comment, marked as answer]
This issue has nothing to do with goreleaser itself, but with the fact you're not running `go get ./...` anywhere in the pipeline.

if you just `git clone` and `go build` your project in a fresh machine it'll fail as well.

**Reply by suntong** [reply]
where is the best place to put the `go get ./...` in the pipeline please?

**Reply by caarlos0** [reply]
Read your output log. 

GoReleaser is not even being called. 

The problem is somewhere else in your pipeline.

**Comment by suntong** [comment]
I wanted to do `go get` myself, but I don't know where should I put that `go get ./...` in the pipeline.
https://github.com/suntong/shuttlebot/actions/runs/888592055/workflow

I've consulted the doc, and put a Global Hooks in `.goreleaser.yml`, where `go generate ./...` normally goes. But it is not working as I expected --
https://github.com/suntong/example/runs/2714335584?check_suite_focus=true

**Comment by suntong** [comment]
I tried to go through the doc again @caarlos0, but am still not be able to find a more proper place than the Global Hooks, which is not working as I expected.

**Reply by caarlos0** [reply]
Its not even failing on the the goreleaser step... 

and, please, reply in the threads so the discussion make sense for future readers.
